### PR TITLE
Add toast notification base component

### PR DESCRIPTION
**Pull request by AnneTee**
I made some adjustments to the existing toast component:
- Renamed it "toast notification" because "toast" can be confusing
- Adjusted the styles so that it's completely independent of the parent component. The toast should be centered within the parent container without needing to add styles to the parent.
- Instead of having to write a CSS rule per-toast to specify animation duration, you can just pass in a duration prop

Basically, Vue made this whole thing much simpler and more flexible. Yay!

### CODE REVIEW

**Comment by AnneTee** [comment]
@egardner I wasn't able to test this on mobile because when I load the page in MobileFrontEnd there's an `Unknown module: vuex` error. Any idea why?

**Comment by egardner** [comment]
I think the mobile errors with Vuex are due to the fact that I forgot to add a `target` line to the module definition in extension.json. I can fix that shortly.

**Comment by egardner** [comment]
Sorry to clutter the history – rebased to get CI tests working, may have added some unnecessary clutter. Will clean up when we squash and merge.

**Comment by AnneTee** [comment]
@egardner I pushed some changes that update the toast component and implement some logic specific to our app in `App.vue` and in the store.

For the toast, when the leave transition happens, a `leave` event is emitted and includes the component key, which I figured might be useful at some point.

For our purposes, I'm only adding a key to the toasts to ensure that a new toast is created each time a success or error message is posted. This is needed in case the user publishes tags quickly – if an existing message is showing and the user clicks publish, a new toast must be created to replace the old one so that the user sees the transition, alerting them of a new message, and so the toast shows for the entire duration.

I updated the shape of the state related to publish status. Though it seems simpler to have `success` and `error` be separate keys in the state, it simplifies the rest of the code a bit to just have a single `publishStatus` key. Let me know what you think about that...

**Review by egardner — APPROVED** [review]
Looks good to me. I think that the way you're integrating the notifications into Vuex here is a good solution. A single `publishStatus` state property is all we need here.

I also like that we can rely on the events already provided by the Transition component here instead of creating new ones. 

The only thing that's potentially a little bit brittle is relying on plain strings to differentiate between success and error states. I think this is fine for now but if we ever needed to get really strict, we could define separate mutations that set `publishStatus` into success vs error states (setPublishError, setPublishSuccess, etc), and then expose getters to components for determining success/error status. This would allow the actual string text of the `publishStatus` states to be more of an internal concern that only specific mutations need to care about.

But let's not add the extra complexity of this approach for now. This is good to merge.

**Comment by AnneTee** [comment]
> The only thing that's potentially a little bit brittle is relying on plain strings to differentiate between success and error states

I completely agree and thought about separating the mutations as a solution. Something to consider in the future if needed.
